cdap-ui/app/directives/metric-picker/metric-picker.html

```diff
@@ -24,17 +24,28 @@
   </div>
 </div>
 
-<div class="input-group">
-  <span class="input-group-addon">
-    <i class="fa fa-line-chart"></i>
-  </span>
-  <input type="text" class="form-control"
-    placeholder="Metric Name"
-    ng-model="metric.name"
-    bs-options="m for m in available.names"
-    data-watch-options="1"
-    data-min-length="0"
-    data-limit="20"
-    data-container="body"
-    bs-typeahead />
+<div ng-repeat="name in metric.names" class="h6">
+  <div class="input-group">
+    <span class="input-group-addon">
+      <i class="fa fa-line-chart"></i>
+    </span>
+    <input type="text" class="form-control"
+      placeholder="Metric Name"
+      ng-model="name.name"
+      bs-options="m for m in available.names"
+      data-watch-options="1"
+      data-min-length="0"
+      data-limit="20"
+      data-container="body"
+      bs-typeahead />
+    <div class="input-group-addon" ng-click="deleteMetric($index)">
+      <span ng-click="deleteMetric($index)">
```

Comment:
We probably don't need this twice. The even bubbles up and this should be caught at the parent.